grin/src/Test/ExtendedSyntax/Grammar.hs

```diff
@@ -0,0 +1,112 @@
+{-# LANGUAGE StandaloneDeriving, DeriveGeneric, LambdaCase #-}
+module Test.ExtendedSyntax.Grammar where
```

Comment:
I think we should remove the old Grammar module as well and we could replace it with the GADT from the mini-grin repo.

Reply:
We should probably do this separately from this PR.

Reply:
Definitely.